dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/validation/ValidationResult.java

```diff
@@ -27,116 +27,15 @@
  */
 package org.hisp.dhis.tracker.validation;
 
-import java.util.ArrayList;
-import java.util.Collections;
 import java.util.List;
-import java.util.function.Predicate;
 
-import lombok.EqualsAndHashCode;
-import lombok.ToString;
-
-@ToString
-@EqualsAndHashCode
-public class ValidationResult
+public interface ValidationResult
```

Comment:
Can you please add some javadocs explaining what its for. As we are exposing this publicly. Would also be great if you can link to it from the package-info.java.

So if someone wants to understand a tracker package they can start by opening package-info.java It states the responsibilities and the main entry point. For validation that is the ValidationService. And maybe what comes out when computations are done in this phase. For validation that is the ValidationResult.